### ISSUE TITLE
False positive for SC2057

### ISSUE THREAD

**Issue by pwagland** [koalaman/shellcheck#1289]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
SC2057

- My shellcheck version (`shellcheck --version` or "online"): 
```sh
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.5.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
if [ \! -d "${SRC_DIR}" ]
then
  echo "could not find the expected source directory: ${SRC_DIR}"
  exit 1
fi
```

#### Here's what shellcheck currently says:

```
Line 2:
if [ \! -d "${SRC_DIR}" ]
        ^-- SC2057: Unknown binary operator.
```

#### Here's what I wanted or expected to see:

Nothing, the `!` operator is allowed, and `\!` should be treated identically, since that is what the shell does.

**Comment by cdoconno** [comment]
This is also the case for `if [\! -f "${FILE_NAME}"]`
#### For bugs
* Rule Id (if any, e.g. SC1000): SC2057
* My shellcheck version(`shellcheck --version` or "online")
```bash
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.7.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

- [X] I tried on shellchec.net and verified this is still a problem on the latest commit 

#### Here's a snippet or screenshot that shows the problem:
```
#!/bin/bash

CONFFILE="$(pwd)/config.conf"
if [ \! -f "$CONFFILE" ]; then
  echo "$SCRIPT: No configuration file found \"$CONFFILE\""
  exit 1
fi
```

#### Here's what shellcheck currently says:
```
Line 4:
if [ \! -f "$CONFFILE" ]; then
        ^-- SC2057: Unknown binary operator.
```

This appears to be an issue with both `-f` and `-d` to check for existence of files or directories
